wgpu-hal/src/gles/adapter.rs

```diff
@@ -713,10 +713,12 @@ impl crate::Adapter<super::Api> for super::Adapter {
                     | Tfc::MULTISAMPLE_X16
             } else if max_samples >= 8 {
                 Tfc::MULTISAMPLE_X2 | Tfc::MULTISAMPLE_X4 | Tfc::MULTISAMPLE_X8
-            } else if max_samples >= 4 {
-                Tfc::MULTISAMPLE_X2 | Tfc::MULTISAMPLE_X4
             } else {
-                Tfc::MULTISAMPLE_X2
+                // The lowest supported level in GLE3.0/WebGL2 is 4X
+                // (see GL_MAX_SAMPLES in https://registry.khronos.org/OpenGL-Refpages/es3.0/html/glGet.xhtml).
+                // On some platforms, like iOS Safari, `get_parameter_i32(MAX_SAMPLES)` returns 0,
+                // so we always fall back to supporting 4x here.
+                Tfc::MULTISAMPLE_X2 | Tfc::MULTISAMPLE_X4
```

Comment:
2X msaa should not be included - support of 4x does not imply 2x and given that e.g. Metal doesn't expose 2x msaa it's not unlikely that it is unsupported

Reply:
correction: it does expose 2x, but not on all devices, ironically on all mobiles though
https://developer.apple.com/documentation/metal/mtldevice/1433355-supportstexturesamplecount

Reply:
should it be removed from all branches then? Sounds like a separate bug fix from this PR though?

Reply:
I don't want to introduce a regression for others who are currently using wgpu with 2x MSAA. 2x MSAA works on my iOS, for instance.